flytekit/bin/entrypoint.py

```diff
@@ -107,7 +107,7 @@ def _dispatch_execute(
         if inspect.iscoroutine(outputs):
             # Handle eager-mode (async) tasks
             logger.info("Output is a coroutine")
-            outputs = asyncio.run(outputs)
+            outputs = asyncio.get_event_loop().run_until_complete(outputs)
```

Comment:
Is there a good way to add a unit test for this?

Reply:
I think `get_running_loop` is preferred over `get_event_loop`, can we try with that?

Reply:
see [comment](https://github.com/python/cpython/issues/96377#issuecomment-1258819222) here for more information, but it's the more thread-safe variant... doesn't really come into play here, but i think better practice.